src/drive/ducks/upload/index.js

```diff
@@ -120,3 +121,62 @@ export const getUploadQueue = state => state[SLUG].queue
 export const getProcessed = state =>
   getUploadQueue(state).filter(f => f.status !== PENDING)
 export const getSuccessful = state => getLoaded(getUploadQueue(state))
+
+// DnD helpers for folder upload
+export const canHandleFolders = evt => {
```

Comment:
It might be easier to test this function if it took the `dataTransfer' object directly instead of an event, wouldn't it?

Reply:
That's what I wanted to do initially but hten I realized it would make `FolderView` handling some of the sordid DOM implementation details I tried to isolate, that's why I decided to pass the event object instead. Either way, I don't think that kind of DOM-oriented code is really testable sadly...